turbinia/workers/volatility.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for executing volatility."""
+
+from __future__ import unicode_literals
+
+import os
+
+from turbinia.evidence import TextFile
+from turbinia.workers import TurbiniaTask
+
+
+class VolatilityTask(TurbiniaTask):
+  """Task to execute volatility."""
+
+  def __init__(self, module='test'):
+    super(VolatilityTask, self).__init__()
+    self.module = module
+
+  def run(self, evidence, result):
+    """Task to execute volatility."""
+
+    # Create the new Evidence object that will be generated by this Task.
+    output_evidence = TextFile()
+    # Create a path that we can write the new file to.
+    # base_name = os.path.basename(evidence.local_path)
+    output_file_path = os.path.join(
+        self.output_dir, '{0:s}.txt'.format(self.id))
+
+    output_evidence.local_path = output_file_path
+
+    # TODO: Add in config options for Turbinia
+    cmd = 'python2 /bin/vol -f {0:s} --profile={1:s} {2:s} \
+            --output=text --output-file={3:s}'.format(
```

Comment:
One other comment here:  Is there an additional log file that can be generated with volatility?   If so, see the log-file related tip here for some suggestions on how to set that:
https://github.com/google/turbinia/blob/master/docs/developing-new-tasks.md#tips

Reply:
From the Volatility options there isn't, although there is a debug output? There's no options to writing this to a specific file, but I could redirect stderr to the log file? (This feels a little hacky, but works fine)

Reply:
Ok, we can leave gathering stdout/stderr to the `TurbiniaTask.execute()` method then.  Maybe just add the debug flag when `config.DEBUG_TASKS` is set then?

Reply:
I've added the boilerplate for this, but there's no way to capture stderr currently so I'm unable to output this to the debug log. I had a quick look, but I don't have the bandwidth to change the subprocess Popen and test all of the modules.